playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java

```diff
@@ -1071,15 +1071,17 @@ public void onTracksInfoReady(PKTracks tracksReady) {
                 //when the track info is ready, cache it in ExoPlayerWrapper. And send event that tracks are available.
                 tracks = tracksReady;
                 shouldRestorePlayerToPreviousState = false;
-                sendDistinctEvent(PlayerEvent.Type.TRACKS_AVAILABLE);
+
                 if (!preferredLanguageWasSelected) {
-                    selectPreferredTracksLanguage();
+                    selectPreferredTracksLanguage(tracksReady);
                     preferredLanguageWasSelected = true;
                 }
 
                 if (tracksReady.getVideoTracks().size() == 0) {
                     exoPlayerView.hideVideoSurface();
                 }
+
+                sendDistinctEvent(PlayerEvent.Type.TRACKS_AVAILABLE);
```

Comment:
we should call this immediately after the conditional if `selectPreferredTracksLanguage`